build.gradle.kts

```diff
@@ -0,0 +1,286 @@
+    dynamicSnapshotVersion.set(false)
+}
+
+fun checkArchiveManifest(archive: File) {
+    ZipFile(archive).use { zip ->
+        zip.getInputStream(zip.getEntry("META-INF/MANIFEST.MF")).bufferedReader().use {
+            check(it.readText().contains("Plugin-Version: ${project.version}")) {
+                "Wrong metadata in file $archive - run a clean build"
+            }
+        }
+    }
+}
+
+tasks.withType<AbstractArchiveTask> {
+    inputs.property("pluginVersion") {
+        project.version
+    }
+}
+
+val jar by tasks.getting(Jar::class)
+val jpi by tasks.getting(War::class)
+
+val checkArchiveManifests: Task by tasks.creating {
+    dependsOn(jar, jpi)
+    doLast {
+        checkArchiveManifest(jar.archiveFile.get().asFile)
+        checkArchiveManifest(jpi.archiveFile.get().asFile)
+    }
+}
+
+tasks.withType<AbstractPublishToMaven> {
+    dependsOn(checkArchiveManifests)
+}
+
+defaultTasks.add("test")
+defaultTasks.add("jpi")
+
+val generateVersionFile: Task by tasks.creating {
+    val versionsDir = layout.buildDirectory.dir("versions")
+    doLast {
+        val outputDir = versionsDir.get().asFile
+        outputDir.mkdirs()
+
+        File(outputDir, "gradle-enterprise-maven-extension-version.txt")
+            .writeText(gradleEnterpriseMavenExtensionVersion)
+        File(outputDir, "common-custom-user-data-maven-extension-version.txt")
+            .writeText(commonCustomUserDataMavenExtensionVersion)
+    }
+
+    inputs.property("gradleEnterpriseMavenExtensionVersion", gradleEnterpriseMavenExtensionVersion)
+    inputs.property("commonCustomUserDataMavenExtensionVersion", commonCustomUserDataMavenExtensionVersion)
+
+    outputs.dir(versionsDir)
+        .withPropertyName("versionsDir")
+}
+
+val createWrapperZip by tasks.creating(Zip::class) {
+    archiveFileName.set("wrapper.zip")
+    destinationDirectory.set(File(test.output.resourcesDir, "gradle"))
+
+    from(project.rootDir) {
+        include("gradle/**")
+        include("gradlew*")
+    }
+}
+
+tasks.processTestResources {
+    dependsOn(createWrapperZip)
+
+    from(includedLibs) {
+        into("hudson/plugins/gradle/injection")
+    }
+    from(generateVersionFile.outputs.files) {
+        into("versions")
+    }
+}
+
+tasks.processResources {
+    filesMatching("hudson/plugins/gradle/injection/InjectionConfig/help-injectMavenExtension.html") {
+        expand("gradleEnterpriseMavenExtensionVersion" to gradleEnterpriseMavenExtensionVersion)
+    }
+    filesMatching("hudson/plugins/gradle/injection/InjectionConfig/help-injectCcudExtension.html") {
+        expand("commonCustomUserDataMavenExtensionVersion" to commonCustomUserDataMavenExtensionVersion)
+    }
+}
+
+tasks.jar {
+    from(includedLibs) {
+        into("hudson/plugins/gradle/injection")
+    }
+    from(generateVersionFile.outputs.files) {
+        into("versions")
+    }
+}
+
+artifacts {
+    add(gradlePluginJpi.name, jpi)
```

Comment:
```suggestion
    gradlePluginJpi(jpi)
```

Does this work?

Reply:
No, unfortunately, it's not possible in Kotlin DSL.